src/preflight/internal-links.js

```diff
@@ -48,9 +48,7 @@ export async function runInternalLinkChecks(scrapedObjects, pageAuthToken, conte
           try {
             const res = await fetch(href, {
               method: 'HEAD',
-              headers: {
-                Authorization: pageAuthToken,
-              },
+              ...options,
```

Comment:
See my above comment - IMO, we should explicitly set the Authorization header if the token is provided via `options`.